File: Data Structures & Algorithms/non-overlapping-intervals/submission-1.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        if len(intervals) <= 1:
            return 0

        intervals.sort(key=lambda x: x[0])

        res = 0
        curr_end = intervals[0][1]
        for i in range(1, len(intervals)):
            start, end = intervals[i]
            if start < curr_end:
                # "Remove" the one that ends later
                if end < curr_end:
                    curr_end = end
                res += 1
            else:
                curr_end = end
        
        return res

File: Data Structures & Algorithms/daily-temperatures/submission-1.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> res(temperatures.size(), 0);

        for (int i = static_cast<int>(temperatures.size()) - 2; i >= 0; --i) {
            auto j = i + 1;
            while (temperatures[i] >= temperatures[j]) {
                if (res[j] == 0) break;
                j = j + res[j];
            }
            if (temperatures[i] < temperatures[j]) res[i] = j - i;
        }

        return res;
    }
};
